triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala

```diff
@@ -363,11 +406,21 @@ class ServiceTest extends AsyncFlatSpec with Eventually with Matchers with Postg
         // trigger.
         _ <- Future { ledgerProxy.disable() }
         _ <- Future { ledgerProxy.enable() }
-        // In the future I hope to be able to prove right here that the
-        // running trigger was failed and restarted. For now, the logs
-        // confirm that to be the case. In any case, the postcondition
-        // is this : the running trigger survives.
+        // To conclude, check that the trigger survived the network
+        // outage and that its history indicates it went through a
+        // restart to do so.
         _ <- assertTriggerIds(uri, "Alice", _ == Vector(aliceTrigger))
+        _ <- assertTriggerStatus(
+          uri,
+          aliceTrigger,
+          _ ==
+            Vector(
+              "starting",
+              "running",
+              "stopped: runtime failure",
```

Comment:
It would be nice to also have a test for a trigger that calls `error`.

Reply:
In order that you see similar results to that you mean? The idea is you'd like to see an execution failure that has manifested from the user script itself?

Reply:
Exactly it should behave basically the same.

Reply:
Writing such a test for highl-level triggers exposed a problem. See PR https://github.com/digital-asset/daml/pull/6202 for the current best running resolution to that. Low-level tests next and we'll see how it goes.

Reply:
Low-level test seems to work. Included with PR https://github.com/digital-asset/daml/pull/6202 as well.